{"links": {"self": {"href": "data/repositories/labscript_suite/lyse/pullrequests/22/comments/41484333.json"}, "html": {"href": "#!/labscript_suite/lyse/pull-requests/22/_/diff#comment-41484333"}}, "parent": {"id": 41484167, "links": {"self": {"href": "data/repositories/labscript_suite/lyse/pullrequests/22/comments/41484167.json"}, "html": {"href": "#!/labscript_suite/lyse/pull-requests/22/_/diff#comment-41484167"}}}, "deleted": false, "pullrequest": {"type": "pullrequest", "id": 22, "links": {"self": {"href": "data/repositories/labscript_suite/lyse/pullrequests/22.json"}, "html": {"href": "#!/labscript_suite/lyse/pull-requests/22"}}, "title": "Unicode strings and other Python 3 forward-compatible changes."}, "content": {"raw": "```\n#!python\n\n>>> sys.getdefaultencoding()\n'ascii'\n>>> sys.getfilesystemencoding()\n'utf-8'\n```\nMh I'm doing some testing of how this could be fixed  my self but nothing found by now.\n\n\nOk I tried this instead of the compile stuff:\n\n\n```\n#!python\n\nexec(open(self.filepath).read(), sandbox, sandbox)\n```\nand still had errors but think these could lead us to the real problem:\n\n\n```\n#!python\n\nTraceback (most recent call last):\n  File \"<string>\", line 13, in <module>\n  File \"/Users/janwerkmann/labscript_suite/lyse/__init__.py\", line 109, in __init__\n    self.group = os.path.basename(__file__).split('.py')[0]\nUnicodeDecodeError: 'ascii' codec can't decode byte 0xcc in position 1: ordinal not in range(128)\n```", "markup": "markdown", "html": "<div class=\"codehilite language-python\"><pre><span></span><span class=\"o\">&gt;&gt;&gt;</span> <span class=\"n\">sys</span><span class=\"o\">.</span><span class=\"n\">getdefaultencoding</span><span class=\"p\">()</span>\n<span class=\"s1\">&#39;ascii&#39;</span>\n<span class=\"o\">&gt;&gt;&gt;</span> <span class=\"n\">sys</span><span class=\"o\">.</span><span class=\"n\">getfilesystemencoding</span><span class=\"p\">()</span>\n<span class=\"s1\">&#39;utf-8&#39;</span>\n</pre></div>\n\n\n<p>Mh I'm doing some testing of how this could be fixed  my self but nothing found by now.</p>\n<p>Ok I tried this instead of the compile stuff:</p>\n<div class=\"codehilite language-python\"><pre><span></span><span class=\"k\">exec</span><span class=\"p\">(</span><span class=\"nb\">open</span><span class=\"p\">(</span><span class=\"bp\">self</span><span class=\"o\">.</span><span class=\"n\">filepath</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">read</span><span class=\"p\">(),</span> <span class=\"n\">sandbox</span><span class=\"p\">,</span> <span class=\"n\">sandbox</span><span class=\"p\">)</span>\n</pre></div>\n\n\n<p>and still had errors but think these could lead us to the real problem:</p>\n<div class=\"codehilite language-python\"><pre><span></span><span class=\"n\">Traceback</span> <span class=\"p\">(</span><span class=\"n\">most</span> <span class=\"n\">recent</span> <span class=\"n\">call</span> <span class=\"n\">last</span><span class=\"p\">):</span>\n  <span class=\"n\">File</span> <span class=\"s2\">&quot;&lt;string&gt;&quot;</span><span class=\"p\">,</span> <span class=\"n\">line</span> <span class=\"mi\">13</span><span class=\"p\">,</span> <span class=\"ow\">in</span> <span class=\"o\">&lt;</span><span class=\"n\">module</span><span class=\"o\">&gt;</span>\n  <span class=\"n\">File</span> <span class=\"s2\">&quot;/Users/janwerkmann/labscript_suite/lyse/__init__.py&quot;</span><span class=\"p\">,</span> <span class=\"n\">line</span> <span class=\"mi\">109</span><span class=\"p\">,</span> <span class=\"ow\">in</span> <span class=\"fm\">__init__</span>\n    <span class=\"bp\">self</span><span class=\"o\">.</span><span class=\"n\">group</span> <span class=\"o\">=</span> <span class=\"n\">os</span><span class=\"o\">.</span><span class=\"n\">path</span><span class=\"o\">.</span><span class=\"n\">basename</span><span class=\"p\">(</span><span class=\"vm\">__file__</span><span class=\"p\">)</span><span class=\"o\">.</span><span class=\"n\">split</span><span class=\"p\">(</span><span class=\"s1\">&#39;.py&#39;</span><span class=\"p\">)[</span><span class=\"mi\">0</span><span class=\"p\">]</span>\n<span class=\"ne\">UnicodeDecodeError</span><span class=\"p\">:</span> <span class=\"s1\">&#39;ascii&#39;</span> <span class=\"n\">codec</span> <span class=\"n\">can</span><span class=\"s1\">&#39;t decode byte 0xcc in position 1: ordinal not in range(128)</span>\n</pre></div>", "type": "rendered"}, "created_on": "2017-07-25T21:19:50.405966+00:00", "user": {"display_name": "Jan Werkmann", "uuid": "{44c4905c-2b90-4045-a5f1-652b8e228626}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D"}, "html": {"href": "https://bitbucket.org/%7B44c4905c-2b90-4045-a5f1-652b8e228626%7D/"}, "avatar": {"href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/557058:a70cc9cf-684e-4849-a61a-9ade4d7218b5/07e5095a-4741-4dc0-a462-9c7d455f961d/128"}}, "nickname": "PhyNerd", "type": "user", "account_id": "557058:a70cc9cf-684e-4849-a61a-9ade4d7218b5"}, "updated_on": "2017-07-25T21:25:18.421147+00:00", "type": "pullrequest_comment", "id": 41484333}